server/src/main.rs

```diff
@@ -5,16 +5,51 @@ mod service;
 mod service_manager;
 mod session;
 
+use clap::Parser;
+use oo7::portal::Secret;
 use service::{Result, Service};
 
 const BINARY_NAME: &str = env!("CARGO_BIN_NAME");
 
+#[derive(Parser)]
+#[command(version, about, long_about = None)]
+struct Args {
+    #[arg(
+        short = 'l',
+        long,
+        default_value_t = false,
+        help = "Read a password from stdin, and use it to unlock the login keyring."
+    )]
+    login: bool,
+}
+
 #[tokio::main]
 async fn main() -> Result<()> {
+    let args = Args::parse();
+    let mut secret: Option<Secret> = None;
     tracing_subscriber::fmt::init();
+
+    if args.login {
+        match rpassword::prompt_password("Enter the login password: ") {
+            Ok(password) => {
+                if password.is_empty() {
+                    tracing::error!("Login password can't be empty.");
+                    std::process::exit(1);
```

Comment:
there is no point in calling exit, if you have an error and you call ? it will automatically unwrap if it is an error.

Reply:
Basically like this
```rust
let password = rpassword::prompt_password("Enter the login password: ")?;
if password.is_empty() {
    return Err(Error::NewErrorTypeThatYouShouldAdd);
}
let secret = Secret::from(password.into_bytes());
```

Reply:
Understood, thanks :)